### PR TITLE
[skip ci] Correct misleading error message

### LINKED ISSUE #2840: Misleading error message when we run out of resources on ESXi
I run out of memory in my nested ESXi while doing this

```
[conur@vir:~] for i in $(seq 10); do echo $i; docker run -it busybox /bin/date; done
1
Fri Oct 21 00:42:50 UTC 2016
2
Fri Oct 21 00:43:02 UTC 2016
3
Fri Oct 21 00:43:15 UTC 2016
4
Fri Oct 21 00:43:27 UTC 2016
5
Fri Oct 21 00:43:40 UTC 2016
6
Fri Oct 21 00:43:51 UTC 2016
7
Fri Oct 21 00:44:04 UTC 2016
8
Fri Oct 21 00:44:16 UTC 2016
9
10
Unable to find image 'busybox:latest' locally
Pulling from library/busybox
56bec22e3559: Already exists
a3ed95caeb02: Already exists
Digest: sha256:98a0bd48d22ff96ca23bfda2fe1cf72034ea803bd79e64a5a5f274aca0f9c51c
Status: Image is up to date for library/busybox:latest
docker: Error: No such image: busybox.
See 'docker run --help'.
```

portlayer shows following
[logs (2).zip](https://github.com/vmware/vic/files/543250/logs.2.zip)

```
time=2016-10-17T14:50:58.646339192Z level=debug msg=[ END ] [github.com/vmware/vic/lib/spec.(*VirtualMachineConfigSpec).AddVirtualCdrom:65] [394.301µs] 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 
time=2016-10-17T14:50:58.646417403Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.(*Handle).Create:218] [6.549829ms] Handle.Create 
time=2016-10-17T14:50:58.646530070Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CreateHandler:68] [20.147349ms]  
time=2016-10-17T14:50:58.647758309Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ScopesHandlersImpl).ScopesAddContainer:191] handle(e3adcc4bea03308d4a34b276bf6ea007) 
time=2016-10-17T14:50:58.647825701Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/network.(*Context).AddContainer:865] 
time=2016-10-17T14:50:58.647866767Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/network.(*Context).findScopes:461] 
time=2016-10-17T14:50:58.647969539Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/network.(*Context).findScopes:461] [102.222µs]  
time=2016-10-17T14:50:58.648100840Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/network.(*Context).AddContainer:865] [275.83µs]  
time=2016-10-17T14:50:58.648189253Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ScopesHandlersImpl).ScopesAddContainer:191] [432.052µs] handle(e3adcc4bea03308d4a34b276bf6ea007) 
time=2016-10-17T14:50:58.648716644Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*InteractionHandlersImpl).JoinHandler:73] 
time=2016-10-17T14:50:58.648779989Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.HandleFromInterface:122] 
time=2016-10-17T14:50:58.648884400Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.HandleFromInterface:122] [104.878µs]  
time=2016-10-17T14:50:58.648976213Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/attach.Join:103] 
time=2016-10-17T14:50:58.649027892Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/attach.Join:103] [51.634µs]  
time=2016-10-17T14:50:58.649218404Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.ReferenceFromHandle:134] 
time=2016-10-17T14:50:58.649278734Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.ReferenceFromHandle:134] [60.819µs]  
time=2016-10-17T14:50:58.649321880Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*InteractionHandlersImpl).JoinHandler:73] [635.386µs]  
time=2016-10-17T14:50:58.650022181Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*LoggingHandlersImpl).JoinHandler:42] 
time=2016-10-17T14:50:58.650073165Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.HandleFromInterface:122] 
time=2016-10-17T14:50:58.650107961Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.HandleFromInterface:122] [35.579µs]  
time=2016-10-17T14:50:58.650210160Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/logging.Join:27] 
time=2016-10-17T14:50:58.650377988Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/spec.(*VirtualMachineConfigSpec).VMPathName:241] 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 
time=2016-10-17T14:50:58.650415571Z level=debug msg=[ END ] [github.com/vmware/vic/lib/spec.(*VirtualMachineConfigSpec).VMPathName:241] [39.525µs] 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 
time=2016-10-17T14:50:58.650479988Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/logging.Join:27] [271.804µs]  
time=2016-10-17T14:50:58.650619378Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.ReferenceFromHandle:134] 
time=2016-10-17T14:50:58.650650883Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.ReferenceFromHandle:134] [31.235µs]  
time=2016-10-17T14:50:58.650760477Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*LoggingHandlersImpl).JoinHandler:42] [725.596µs]  
time=2016-10-17T14:50:58.651751448Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CommitHandler:212] handle(e3adcc4bea03308d4a34b276bf6ea007) 
time=2016-10-17T14:50:58.652475056Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Commit:273] 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 
time="2016-10-17T14:50:58Z" level=error msg="task failed: A value exceeded the acceptable range." 
time="2016-10-17T14:50:58Z" level=error msg="Something failed. Spec was {DynamicData:{} ChangeVersion: Name:dreamy_almeida-53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 Version: Uuid:db6f7adf-cf45-59e0-b74a-d47dfd8500dc InstanceUuid: NpivNodeWorldWideName:[] NpivPortWorldWideName:[] NpivWorldWideNameType: NpivDesiredNodeWwns:0 NpivDesiredPortWwns:0 NpivTemporaryDisabled:<nil> NpivOnNonRdmDisks:<nil> NpivWorldWideNameOp: LocationId: GuestId:other3xLinux64Guest AlternateGuestName: Annotation: Files:0xc4206cbd60 Tools:<nil> Flags:<nil> ConsolePreferences:<nil> PowerOpInfo:<nil> NumCPUs:2 NumCoresPerSocket:0 MemoryMB:2048 MemoryHotAddEnabled:0xc420811260 CpuHotAddEnabled:0xc420811260 CpuHotRemoveEnabled:<nil> VirtualICH7MPresent:<nil> VirtualSMCPresent:<nil> DeviceChange:[0xc4207b61e0 0xc4207b6500 0xc4207b6820 0xc4207b6a00 0xc4207b6d70 0xc4207b6f00 0xc4207b7270 0xc4207b7310] CpuAllocation:<nil> MemoryAllocation:<nil> LatencySensitivity:<nil> CpuAffinity:<nil> MemoryAffinity:<nil> NetworkShaper:<nil> CpuFeatureMask:[] ExtraConfig:[0xc4205dda00 0xc4205dda20 0xc4205dda40 0xc4205dda80 0xc4205ddaa0 0xc4205ddac0 0xc4205ddae0 0xc4205ddb00 0xc4205ddb20 0xc4205ddb60 0xc4205c6e80 0xc4205c6ea0 0xc4205c6ec0 0xc4205c6ee0 0xc4205c6f20 0xc4205c6f40 0xc4205c6f60 0xc4205c6f80 0xc4205c6fe0 0xc4205c7000 0xc4205c7020 0xc4205c7040 0xc4205c7080 0xc4205c70c0 0xc4205c7100 0xc4205c7120 0xc4205c7160 0xc4205c71a0 0xc4205c71c0 0xc4205c71e0 0xc4205c7200 0xc4205c7220 0xc4205c7260 0xc4205c7280 0xc4205c72a0 0xc4205c72c0 0xc4205c7300 0xc4205c7340 0xc4205c7360 0xc4205c7560 0xc4205c7580 0xc4205c7600 0xc4205c7620 0xc4205c7640 0xc4205c7660 0xc4205c76a0 0xc4205c76c0 0xc4205c76e0 0xc4205c7700 0xc4206be660 0xc4206be680 0xc4206be6a0 0xc4206be6c0 0xc4206be6e0 0xc4206be700 0xc4206be740 0xc4206be760 0xc4206be7a0 0xc4206be7c0 0xc4206be860 0xc4206be8a0 0xc4206be8e0 0xc4206be900 0xc4206be980 0xc4206be9a0 0xc4206be9e0 0xc4206bea00 0xc4206bea20 0xc4206bea40 0xc4206bea80 0xc4206beac0 0xc4206beb20 0xc4206beb60 0xc4206beba0 0xc4206bebe0 0xc4206bec00 0xc4206bec20 0xc4206bec40 0xc4206bec60 0xc4206bec80 0xc4206beca0 0xc4206becc0 0xc4206bed00 0xc4206bed20 0xc4206bed40 0xc4206bed60 0xc4206bed80 0xc4206beda0 0xc4206bedc0 0xc4206bede0 0xc4206bee20 0xc4206bee40 0xc4206bee60 0xc4206bee80 0xc4206beea0 0xc4206beec0 0xc4206beee0 0xc4206bef20 0xc4206bef40 0xc4206bef80 0xc4206befa0 0xc4206befc0 0xc4206befe0] SwapPlacement: BootOptions:<nil> VAppConfig:<nil> FtInfo:<nil> RepConfig:<nil> VAppConfigRemoved:<nil> VAssertsEnabled:<nil> ChangeTrackingEnabled:<nil> Firmware: MaxMksConnections:0 GuestAutoLockEnabled:<nil> ManagedBy:<nil> MemoryReservationLockedToMax:<nil> NestedHVEnabled:<nil> VPMCEnabled:<nil> ScheduledHardwareUpgradeInfo:<nil> VmProfile:[] MessageBusTunnelEnabled:<nil>}" 
time="2016-10-17T14:50:58Z" level=debug msg="Committing container 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 status as: " 
time=2016-10-17T14:50:58.693964772Z level=debug msg=[ END ] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Commit:273] [41.489357ms] 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213 
time="2016-10-17T14:50:58Z" level=error msg="CommitHandler error on handle(e3adcc4bea03308d4a34b276bf6ea007) for 53d3c2239453955485a5a9421a6a0a1614e5b6be32371523284c70096abf4213: task.Error{LocalizedMethodFault:(*types.LocalizedMethodFault)(0xc4207c2660)}" 
time=2016-10-17T14:50:58.693982870Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CommitHandler:212] [42.233308ms] handle(e3adcc4bea03308d4a34b276bf6ea007) 
```

### PR DESCRIPTION
**Pull request by sgairo**
Fixes #2840

### CODE REVIEW

File: lib/portlayer/exec/commit.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func Commit(ctx context.Context, sess *session.Session, h *Handle, waitTime *int
 		}
 
 		if err != nil {
-			log.Errorf("Something failed. Spec was %+v", *h.Spec.Spec())
+			log.Errorf("Waiting for operations to complete failed: %s. Spec was %+v", err, *h.Spec.Spec())
```
Comment (emlin) [inline lib/portlayer/exec/commit.go]: this is the error to create container vm, should we make it clearer?

Comment (sgairo) [inline lib/portlayer/exec/commit.go]: Yes, do you think it should say failed to create container vm instead ?

Comment (matthewavery) [inline lib/portlayer/exec/commit.go]: perhaps `An error occurred while waiting for a creation operation to complete...` :-) that way someone might be able to zip right to this section of the code for commit when they see this error.

Comment (sgairo) [inline lib/portlayer/exec/commit.go]: i like that combo, will do!

**Comment by caglar10ur** [comment]
LGTM

**Comment by emlin** [comment]
lgtm

**Pull request merged**